web/modules/custom/user_timezone/src/Form/UserTimezoneSettingForm.php

```diff
@@ -0,0 +1,168 @@
+    ];
+
+    $form['morning_end'] = [
+      '#type' => 'number',
+      '#title' => $this->t('End of morning'),
+      '#description' => $this->t('Set the end point of your custom morning time range (e.g.: 06-12)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('morning_end'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['afternoon_start'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Beginning of afternoon'),
+      '#description' => $this->t('Set the start point of your custom afternoon time range (e.g.: 12-18)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('afternoon_start'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['afternoon_end'] = [
+      '#type' => 'number',
+      '#title' => $this->t('End of afternoon'),
+      '#description' => $this->t('Set the end point of your custom afternoon time range (e.g.: 12-18)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('afternoon_end'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['evening_start'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Beginning of evening'),
+      '#description' => $this->t('Set the start point of your custom evening time range (e.g.: 18-00)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('evening_start'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['evening_end'] = [
+      '#type' => 'number',
+      '#title' => $this->t('End of evening'),
+      '#description' => $this->t('Set the end point of your custom evening time range (e.g.: 18-00)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('evening_end'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['night_start'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Beginning of night'),
+      '#description' => $this->t('Set the start point of your custom night time range (e.g.: 00-05)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('night_beginning'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    $form['night_end'] = [
+      '#type' => 'number',
+      '#title' => $this->t('End of night'),
+      '#description' => $this->t('Set the end point of your custom night time ranger (e.g.: 00-05)'),
+      '#min' => 00,
+      '#max' => 23,
+      '#default value' => $config->get('night_end'),
+      '#element validate' => 'validateNumber',
+      '#required' => TRUE,
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
```

Comment:
The "ends" should end later than the "starts".

It would be nice to have a validation that adds together the times, to make sure that all 24 hrs are covered.

I am also thinking about that maybe the "end" times are unnecessary, since they are kind of implied with the "start" times...